propkatraj/propkatraj.py

```diff
@@ -152,7 +151,10 @@ class PropkaTraj(AnalysisBase):
     def __init__(self, atomgroup, select='protein', skip_failure=False,
                  **kwargs):
         if select is not None:
-            self.ag = atomgroup.select_atoms(select).atoms
+            if not isinstance(select, str):
+                self.ag = atomgroup.atoms[select].atoms
```

Comment:
Why is this necessary? Can `select` be a index array??

Reply:
The doc string says that it can only be a string. If you want it to also be an index array then you need to change the docs and a versionchanged.

However, MDAnalysis analysis functions never use index arrays: either a selection or an AtomGroup. I'd probably just remove the ability to pass anything except a selection string and keep it simple.

Reply:
Agreed, I'm also opposed to using an index array unless there's a good reason for it.